src/pages/ProductDetail.jsx

```diff
@@ -0,0 +1,91 @@
+import React from 'react';
+import { useParams } from 'react-router-dom';
+import style from './ProductDetail.module.css';
+import { getProductDetail } from '../data/API';
+import { useState, useEffect } from 'react';
+import Button from '../components/ui/button/Button';
+import { addOrUpdateToCart } from '../api/firebase';
+import { useNavigate } from 'react-router-dom';
+import { useRecoilState } from 'recoil';
+import { loginState, userInfoState } from '../data/LoginData';
+import { BsCartX } from 'react-icons/bs';
+
+const SHIPPING = 3000;
+export default function ProductDetail() {
+  const [isLoggedIn, setIsLoggedIn] = useRecoilState(loginState);
+  const [userInfo, setUserInfo] = useRecoilState(userInfoState);
+  const userName = userInfo.displayName;
+  const { id } = useParams();
+  const navigate = useNavigate();
+  const [detail, setDetail] = useState([]);
+  const { id: productId, title, photo, price, description, isSoldOut } = detail;
+
+  useEffect(() => {
+    const details = getProductDetail(id);
+    details.then((data) => {
+      setDetail(data);
+    });
+  }, [id]);
+
+  //최근 본 상품 로컬스토리지 저장 용도
+  useEffect(() => {
+    if (localStorage.watched === undefined) {
+      localStorage.setItem('watched', JSON.stringify([]));
+    }
```

Comment:
```suggestion
    if ( !localStorage.watched ) localStorage.setItem('watched', JSON.stringify([]));
```
굳이 === undefined 는 안적어도 될거 같네요